packages/server/graphql/public/mutations/persistIntegrationSearchQuery.ts

```diff
@@ -0,0 +1,68 @@
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getIntegrationProvidersByIds from '../../../postgres/queries/getIntegrationProvidersByIds'
+import upsertIntegrationSearchQuery from '../../../postgres/queries/upsertIntegrationSearchQuery'
+import {getUserId, isTeamMember} from '../../../utils/authorization'
+import publish from '../../../utils/publish'
+import {MutationResolvers} from '../resolverTypes'
+
+const persistIntegrationSearchQuery: MutationResolvers['persistIntegrationSearchQuery'] = async (
+  _source,
+  {teamId, service, providerId, jiraServerSearchQuery},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const viewerId = getUserId(authToken)
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  //AUTH
+  if (!isTeamMember(authToken, teamId)) {
+    return {error: {message: `Not on team`}}
+  }
+
+  let query
+  switch (service) {
+    case 'jiraServer':
+      query = jiraServerSearchQuery!
+      break
+    // case "jira":
+    //   query = jiraSearchQuery!
+    // break;
+    default:
+      throw new Error('Service not implemented')
+  }
+
+  if (providerId) {
+    const integrationProviders = await getIntegrationProvidersByIds([providerId])
```

Comment:
+1 could save a line and do `const integrationProvider = await getIntegrationProvidersByIds([providerId])[0]`